neuraxle/pipeline.py

```diff
@@ -158,52 +149,31 @@ def load(self, pipeline: 'Pipeline', data_container: DataContainer) -> Optional[
         :return: pipeline
         """
         pipeline_cache_folder = os.path.join(self.cache_folder, pipeline.name)
-        pipeline_cache_list_file_name_path = os.path.join(pipeline_cache_folder, self.pipeline_cache_list_file_name)
-
-        with open(pipeline_cache_list_file_name_path, mode='r') as file:
-            lines = file.readlines()
 
-            pipeline_path_line_index = self._find_saved_pipeline_path_line_index(
-                data_container=data_container,
-                lines=lines
+        return load(
+            self._create_next_cached_pipeline_path(
+                pipeline_cache_folder,
+                data_container
             )
-            saved_pipeline_path = lines[pipeline_path_line_index].strip()
-
-            return load(saved_pipeline_path)
-
-    @staticmethod
-    def _find_saved_pipeline_path_line_index(data_container, lines):
-        """
-        Find saved pipeline path line
-
-        :param data_container:
-        :param lines:
-        :return:
-        """
-        for index, line in enumerate(lines):
-            if str(data_container) == line.strip():
-                return index + 1
-        return -1
+        )
 
-    def _create_next_cached_pipeline_path(self, pipeline_cache_folder) -> str:
+    def _create_next_cached_pipeline_path(self, pipeline_cache_folder, data_container: 'DataContainer') -> str:
         """
         Create next cached pipeline path by incrementing a suffix
 
-        :param pipeline_cache_folder:
+        :type data_container: DataContainer
+        :param pipeline_cache_folder: str
         :return: path string
         """
-        cached_pipeline_paths = [path for path in glob.glob(os.path.join(pipeline_cache_folder, '*'))]
-        pipeline_file_name_index = 0
-
-        while self._create_cached_pipeline_path(pipeline_cache_folder,
-                                                pipeline_file_name_index) in cached_pipeline_paths:
-            pipeline_file_name_index += 1
-
-        return self._create_cached_pipeline_path(pipeline_cache_folder, pipeline_file_name_index)
+        all_current_ids_hash = None
+        for current_id, *_ in data_container:
+            m = hashlib.md5()
+            m.update(str.encode(current_id))
+            if all_current_ids_hash is not None:
```

Comment:
Doesn't it feels odd to see hashing logic inside a saver? 

Perhaps this logic should be pushed into elsewhere (TDA) by doing `data_container.summarize_hash_all_ids()` here, or otherwise this logic should be computed before and send to here as an other argument (still not a good fix), or perhaps it should be a private method here in the saver. 

Note: I don't really like that the hashing logic be found in the data_container either. That might mean there is a better solution. Could you think of something else and weight yourself the pros and cons? Perhaps a private method here in the saver would do... `self.summarize_hash_all_ids`...?

Reply:
you are commenting commits from another pr. I need the changes from the other pr here (see title of this pr). resolving this comment here.